Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1474,6 +1474,13 @@ void FrameLoader::loadURL(FrameLoadRequest&& frameLoadRequest, const String& ref
     if (m_inStopAllLoaders || m_inClearProvisionalLoadForPolicyCheck)
         return;
 
+#if PLATFORM(WPE) || PLATFORM(GTK)
```

Comment:
What's actually going to happen is we'll forget about this, WPE/GTK will be fixed, and macOS/iOS/*OS will remain broken indefinitely. It should be done for all platforms or not at all.

Reply:
It will be applied for all platforms in the next few moths. That is the plan.

Reply:
Just apply it on all platforms immediately. If it breaks something, somebody will notice and revert. If you make it specific to GTK/WPE, then who knows when we'll notice.

Reply:
(And conversely, if Alex thinks it's too risky to do this on macOS/iOS for whatever reason, then it's also too risky for WPE/GTK.)

Reply:
I don't think it's very risky, but internally this is a very risk-averse time of year and any breakage from this change would be subtle and not noticed for a while.

I do think this needs a unit test, though.

Reply:
My apologies @achristensen07 for wrongly interpreting you in our private chat. 

This change is not compliant with spec: https://html.spec.whatwg.org/multipage/browsing-the-web.html#session-history-traversal-parallel-queue-sync-nav-steps, so I am still working on proper fix but due to complexity of the navigation algorithm (sync fragment navigation vs. async navigation) it is not easy task.